src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,449 @@
+module Algebra.Graph.Acyclic.AdjacencyMap (
+  -- * Data structure
+  AdjacencyMap, aam,
+
+  -- * Basic graph construction primitives
+  empty, vertex, disjointOverlay, disjointConnect, vertices,
+
+  -- * Graph properties
+  isEmpty, edgeList, vertexList, edgeSet, vertexSet,
+  vertexCount, edgeCount, adjacencyList, hasEdge, hasVertex,
+
+  -- * Graph transformation
+  removeVertex, removeEdge, transpose, induce,
+
+  -- * Graph composition
+  box,
+
+  -- * Relational operations
+  transitiveClosure,
+
+  -- * Functions on acyclic graphs
+  topSort,
+
+  -- * Acyclic graph construction methods
+  scc, fromGraph, PartialOrder,
+
+  -- * Miscellaneous
+  consistent,
+  ) where
+
+import Algebra.Graph (Graph, foldg)
+import qualified Algebra.Graph.AdjacencyMap as AM
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AM
+import qualified Algebra.Graph.NonEmpty.AdjacencyMap as NonEmpty
+import qualified Data.Graph.Typed as Typed
+import Data.Set (Set)
+import Data.Coerce (coerce)
+
+{-| The 'AdjacencyMap' data type represents an acyclic graph.
+All the methods provided in this module are safe and cannot be used
+to produce a cyclic graph.
+
+We define a 'Num' instance as a convenient notation for working
+with graphs:
+
+  > vertexList 0           == [0]
+  > edgeList   0           == []
+  > vertexList (1 + 2)     == [1,2]
+  > edgeList   (1 + 2)     == []
+  > vertexList (1 * 2)     == [1,2]
+  > edgeList   (1 * 2)     == [(1,2)]
+  > vertexList (1 + 2 * 3) == [1,2,3]
+  > edgeList   (1 + 2 * 3) == [(2,3)]
+  > vertexList (1 * 2 + 3) == [1,2,3]
+  > edgeList   (1 * 2 + 3) == [(1,2)]
+
+In the `Num` operations, we only keep edges that align with the order
+of integers. Ie. Edges can only be formed from vertex numbered /a/ 
+to a vertex numbered /b/ if and only if /a/ \< /b/:
+
+  > edgeList (1 * 2) == [(1,2)]
+  > edgeList (2 * 1) == []
+  > (2 * 1)          == (2 + 1)
+
+__Note:__ the 'Num' instance does not satisfy several "customary laws"
+of 'Num', which dictate that 'fromInteger' @0@ and 'fromInteger' @1@
+should act as additive and multiplicative identities, and 'negate' as
+additive inverse. Nevertheless, overloading 'fromInteger', '+' and '*'
+is very convenient when working with algebraic graphs; we hope that
+in future Haskell's Prelude will provide a more fine-grained class
+hierarchy for algebraic structures, which we would be able to utilise
+without violating any laws.
+-}
+newtype AdjacencyMap a = AAM
+  { aam :: AM.AdjacencyMap a
+  } deriving (Show, Eq, Ord)
```

Comment:
I don't think we can derive the `Show` instance, because it would use graph construction primitives like `edge`, which are not available in the API here. We'll need to figure out something else, and it looks like a non-trivial task.

Reply:
I cannot write the `API` for `Acyclic.AdjacencyMap` is there isn't a `Show` instance. I am not using the `API` anyway, so should I remove the `API`?